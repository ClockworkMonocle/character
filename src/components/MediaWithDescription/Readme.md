Align items side-by-side, with a slight gap between them, vertically centered.

```js
<MediaWithDescription
  media={<div style={{ height: '100px', width: '100px', backgroundColor: 'purple' }}/>}
  description={[
    <p>heyo</p>,
    <p>this is some really awesome content</p>
  ]}
/>
```
```js
<MediaWithDescription
  description={<p>Just some normal content. No media to balance against</p>}
/>
```
