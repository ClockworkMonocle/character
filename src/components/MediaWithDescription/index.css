
.media_desc--container {
  width: 100%;
  height: 100%;;
  display: flex;
  flex-direction: row;
  align-items: center;
}

.media_desc--media {
  margin-right: 8px;
}

.media_desc--desc {
  flex-grow: 1;
}
