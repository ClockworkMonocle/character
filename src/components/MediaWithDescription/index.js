import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import './index.css';

export default function MediaWithDescription({ media, description, className, ...props }) {
  return (
    <div className={cn('media_desc--container', className)}>
      <div className="media_desc--media">{media}</div>
      <div className="media_desc--desc">{description}</div>
    </div>
  );
}

MediaWithDescription.propTypes = {
  media: PropTypes.node,
  description: PropTypes.node,
};
