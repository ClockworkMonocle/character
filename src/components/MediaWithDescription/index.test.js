import React from 'react';
import { shallow } from 'enzyme';
import MediaWithDescription from './index';

test('<MediaWithDescription /> renders', () => {
  expect(shallow(<MediaWithDescription
    media={<div>media</div>}
    description={<div>description</div>}
  />)).toMatchSnapshot();
});
