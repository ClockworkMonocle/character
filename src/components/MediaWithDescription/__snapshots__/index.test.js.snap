// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaWithDescription /> renders 1`] = `
<div
  className="media_desc--container"
>
  <div
    className="media_desc--media"
  >
    <div>
      media
    </div>
  </div>
  <div
    className="media_desc--desc"
  >
    <div>
      description
    </div>
  </div>
</div>
`;
